### PR TITLE
Add new `kb_document` option to the generate strategy

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description:
Add new `kb_document` option to the generate strategy for adding documentation of the generated instance to the knowledge base. 

Also added a generic settings strategy to configure the knowledge base. The current implementation is independent of oteapi-dlite, but is kept in this repo to have better control over it.

## Type of change:
- [ ] Bug fix.
- [x] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand, including clearly named variables?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: oteapi_dlite/strategies/generate.py [hunk 1]
```diff
@@ -110,6 +115,33 @@ class DLiteStorageConfig(AttrDict):
...
+            ),
+        ),
+    ] = None
+    kb_document_context: Annotated[
+        Optional[dict],
+        Field(
+            description=(
+                "If `kb_document_iri` is given, this configuration adds "
+                "will add additional context to the documentation of the "
+                "generated instance."
```
Comment (francescalb) [inline oteapi_dlite/strategies/generate.py:137]:
```suggestion
                "If `kb_document_iri` is given, this configuration adds "
                "additional context to the documentation of the "
                "generated instance."
```

File: oteapi_dlite/strategies/settings.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class SettingsConfig(AttrDict):
+    """Configuration for a generic "settings" filter.
+
+    This strategy stores settings in the session such that they are
+    available for other strategies later in the pipeline.
+
+    The settings are stored as a JSON string and can be accessed
+    by its label.
```
Comment (francescalb) [inline oteapi_dlite/strategies/settings.py:28]:
```suggestion
    The settings are stored as a JSON string which can be accessed
    by its label.
```

File: oteapi_dlite/strategies/settings.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                "It should be unique."
+            ),
+        ),
+    ]
+    settings: Annotated[
+        Union[dict, list, str, int, float, bool, NoneType],
+        Field(
+            description=(
+                "The configurations to store, represented as a Python "
+                "object that can be serialised to JSON."
```
Comment (francescalb) [inline oteapi_dlite/strategies/settings.py:46]:
```suggestion
                "The configurations to be stored, represented as a Python "
                "object that can be serialised to JSON."
```

File: oteapi_dlite/strategies/generate.py [hunk 4]
```diff
@@ -110,6 +116,58 @@ class DLiteStorageConfig(AttrDict):
...
+                "...     filterType='application/vnd.dlite-settings',\n"
+                "...     configuration={\n"
+                "...         'label': 'tripper.triplestore',\n"
+                "...         'settings': {\n"
+                "...             'backend': 'rdflib',\n"
+                "...             'triplestore_url': '/path/to/local/kb.ttl',\n"
+                "...         },\n"
+                "...     },\n"
+                "... )\n"
+                ">>> generate = client.create_function(...)\n"
```
Comment (francescalb) [inline oteapi_dlite/strategies/generate.py:145]:
```suggestion
                ">>> generate = client.create_function(kb_document_class=IRIOfClassIn_tripper.trippelstore)\n"
```

Comment (jesper-friis) [reply oteapi_dlite/strategies/generate.py:145]: > generate = client.create_function(kb_document_class=IRIOfClassIn_tripper.trippelstore)\n"

`client.create_function()` krever mange fler argumenter enn bare `kb_document_class`. Tanken med eksemplet var bare å vise hvordan settings-strategien skal brukes sammen med generate, ikke å inkludere et fult kjørbart eksempel i beskrivelsen av `kb_document_class`. 

Et slikt eksempel vill bestemt være nyttig, men jeg tror at det burde inkluderes i dokumentasjonen på samme måte som docs/tutorial.md er inkludert i tripper.

**Review by francescalb — APPROVED** [review]
Very nice!

**Pull request merged**